### PR TITLE
ADCM 2772 Edit Host entity to work with maintenance mode

### PR DESCRIPTION
**Pull request by DanBalalan**

### CODE REVIEW

File: tests/functional/test_multiple_adcm.py [hunk 1]
```diff
@@ -93,7 +93,7 @@ def test_export_cluster_from_another_adcm(adcm_fs, extra_adcm_fs, sdk_client_fs,
     Test basic scenario export of a cluster from one ADCM to another
     """
 
-    provider = sdk_client_fs.provider()
+    provider = sdk_client_fs.provider(name=PROVIDER_NAME)
     cluster_to_export = sdk_client_fs.cluster(name=CLUSTER_NAME)
 
     hc_map = set_hc_map(cluster_to_export, provider)
```
Comment (Sealwing) [inline tests/functional/test_multiple_adcm.py:99]: please rebase on `ADCM-2072`, sync with the `develop` branch will come later)

**Review by Sealwing — CHANGES_REQUESTED** [review]

File: python/api/host/views.py [hunk 2]
```diff
@@ -230,6 +231,43 @@ def delete(self, request, *args, **kwargs):
...
+        )
+        if serializer.is_valid(raise_exception=True):
+            self.__check_maintenance_mode_constraint(serializer.validated_data['maintenance_mode'])
+            serializer.save(**kwargs)
+            return Response(self.get_serializer(self.get_object()).data, status=success_status)
+        return Response(serializer.errors, status=onerror_status)
+
+    @staticmethod
+    def __check_maintenance_mode_constraint(new_mode):
+        if new_mode not in (MaintenanceModeType.On.value, MaintenanceModeType.Off.value):
```
Comment (a-alferov) [inline python/api/host/views.py:268]: Add pls check `disabled` value

File: python/api/host/views.py [hunk 3]
```diff
@@ -230,6 +231,43 @@ def delete(self, request, *args, **kwargs):
...
+        serializer = self.serializer_class(
+            host,
+            data=request.data,
+            context={
+                'request': request,
+                'prototype_id': kwargs.get('prototype_id', None),
+                'cluster_id': kwargs.get('cluster_id', None),
+                'provider_id': kwargs.get('provider_id', None),
+            },
+            partial=True,
```
Comment (a-alferov) [inline python/api/host/views.py:258]: I think because of this code `PUT` method will work like `PATCH`

File: python/api/host/serializers.py [hunk 4]
```diff
@@ -31,9 +31,10 @@ class HostSerializer(serializers.Serializer):
     prototype_id = serializers.IntegerField(help_text='id of host type')
     provider_id = serializers.IntegerField()
     fqdn = serializers.CharField(help_text='fully qualified domain name')
-    description = serializers.CharField(required=False)
+    description = serializers.CharField(required=False, allow_blank=True)
     state = serializers.CharField(read_only=True)
     url = ObjectURL(read_only=True, view_name='host-details')
+    maintenance_mode = serializers.ChoiceField(choices=MaintenanceModeType.choices, required=True)
```
Comment (a-alferov) [inline python/api/host/serializers.py:37]: I think `required=True` is redundant. We want this field not to be required when creating an object, but only required when updating

**Review by a-alferov — APPROVED** [review]

**Pull request merged**